### PR TITLE
fix llama3 type rope

### PR DESCRIPTION
**Pull request by hanq-moreh**
## Motivation

The Llama3 type RoPE (Rotary Position Embedding) computation in SpecForge differed from the implementation used in SGLang.
This pull request aligns SpecForge’s RoPE calculation with SGLang’s implementation (see [rotary_embedding.py#L913](https://github.com/moreh-dev/sglang/blob/90401cf7d243da5dcb4f2cef527e2b342ea468b5/python/sglang/srt/layers/rotary_embedding.py#L913)￼).

With this change, SpecForge models can now correctly utilize configurations such as [nvidia/gpt-oss-120b-Eagle3](https://huggingface.co/nvidia/gpt-oss-120b-Eagle3/blob/main/config.json)￼ for training.

## Modifications
- Updated `train_eagle3_offline.py`, `train_eagle3_online.py`, and `train_eagle3_sgl_online.py` to set draft_vocab_size equal to the target model’s vocab size when it is null in the config.
- Modified `llama3_eagle.py` to align Llama3-type RoPE computation with SGLang’s implementation.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://sgl-fru7574.slack.com/archives/C09784E3EN6 to discuss your PR.

### CODE REVIEW

File: scripts/train_eagle3_online.py [hunk 1]
```diff
@@ -313,6 +313,9 @@ def build_draft_model(args: Namespace) -> Tuple[AutoDraftModelConfig, nn.Module]
         # Use provided config file
         draft_model_config = AutoDraftModelConfig.from_file(args.draft_model_config)
 
+    if draft_model_config.draft_vocab_size is None:
+        draft_model_config.draft_vocab_size = draft_model_config.vocab_size
+
```
Comment (FrankLeeeee) [inline scripts/train_eagle3_online.py]: I think this will better put inside `specforge.modeling.auto.AutoDraftModelConfig`, we can handle this logic inside the `from_file` method.

**Comment by FrankLeeeee** [comment]
Can you rebase main so that the CI will trigger the updated workflow?

**Comment by hanq-moreh** [comment]
> Can you rebase main so that the CI will trigger the updated workflow?

Rebased onto the main. Please let me know if you need me to re-run checks or if anything looks off.

**Comment by FrankLeeeee** [comment]
LGTM. Thanks!
